code/__HELPERS/mobs.dm

```diff
@@ -357,4 +385,19 @@ Proc for attack log creation, because really why not
 	if (!mind || !mind.assigned_job)
 		return
 
-	return mind.assigned_job.head_position
\ No newline at end of file
+	return mind.assigned_job.head_position
+
+/mob/proc/get_screen_colour()
+
+/mob/proc/update_client_colour(time = 10) //Update the mob's client.color with an animation the specified time in length.
+	if(!client) //No client_colour without client. If the player logs back in they'll be back through here anyway.
+		return
+	client.colour_transition(get_screen_colour(), time = time) //Get the colour matrix we're going to transition to depending on relevance (magic glasses first, eyes second).
+
+/mob/living/carbon/human/get_screen_colour() //Fetch the colour matrix from wherever (e.g. eyes) so it can be compared to client.color.
+	. = ..()
+	if(.)
+		return .
+	var/obj/item/organ/internal/eyes/eyes = internal_organs_by_name[BP_EYES]
+	if(eyes) //If they're not, check to see if their eyes got one of them there colour matrices. Will be null if eyes are robotic/the mob isn't colourblind and they have no default colour matrix.
+		return eyes.get_colourmatrix()
```

Comment:
Is a null colourmatrix safe?

Reply:
In my tests it has worked well.
Paradise uses a default null colourmatrix , too.